### ISSUE TITLE
After reconnection, subscription seems to be lost

### ISSUE THREAD

**Issue by dbousamra** [giabao/paho-akka#2]
Hi,

If I am connected to a mqtt broker, and then I terminate the server, my program correctly reconnects. And I can correctly send publish messages to the broker. But I cannot receive messages. It fails to resubscribe or something. I am new to mqtt so it could be an mqtt misunderstanding, rather than a bug in paho-akka. Here is some example code:

``` scala
object Example {

  val system = ActorSystem("x", None, None, Some(ExecutionContext.fromExecutor(Executors.newFixedThreadPool(10))))

  val topic = "example-topic"

  val pubsub = system.actorOf(Props(classOf[MqttPubSub], PSConfig(
    brokerUrl = "tcp://localhost:1884", //all params is optional except brokerUrl
    userName = null,
    password = null,
    //messages received when disconnected will be stash. Messages isOverdue after stashTimeToLive will be discard
    stashTimeToLive = 1.minute,
    stashCapacity = 8000, //stash messages will be drop first haft elems when reach this size
    reconnectDelayMin = 10.millis, //for fine tuning re-connection logic
    reconnectDelayMax = 30.seconds,
    cleanSession = false
  )))

  class SubscribeActor extends Actor with LazyLogging {
    pubsub ! Subscribe(topic, self)

    def receive = {
      case SubscribeAck(Subscribe(topic, `self`, _)) => {
        logger.info("Subscription set up")
        context become ready
      }
    }

    def ready: Receive = {
      case msg: Message => logger.info(new String(msg.payload))
    }
  }

  def main(args: Array[String]) {

    val subscriber = system.actorOf(Props(classOf[SubscribeActor]))

    while (true) {
      readLine()
      pubsub ! new Publish(topic, "HELLO WORLD".getBytes)
    }

  }
```

**Comment by aiacovella** [comment]
Is there any resolution to this issue. I'm experiencing the same thing.

**Comment by aiacovella** [comment]
I had experienced the same thing. I've applied changes to this pr: https://github.com/giabao/paho-akka/pull/6 and that seems to fix this issue at least in my case. I'm waiting for the pr to be merged.

**Comment by giabao** [comment]
I'm so sorry for not checking this issue.
I will review #6 now.

**Comment by giabao** [comment]
resolved in [version 1.3.0](../blob/master/CHANGES.md#v130)
